crates/aptos-crypto/src/bls12381/mod.rs

```diff
@@ -0,0 +1,195 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+//! This module provides APIs for Boneh-Lynn-Shacham (BLS) multi-signatures on top of Barreto-Lynn-Scott
+//! BLS12-381 elliptic curves. This module wraps the [blst](https://github.com/supranational/blst)
+//! library.
+//!
+//! More specifically, we build BLS multisignatures as described in [^BLS04], [^Bold03] but using
+//! the proof-of-possession (PoP) scheme from [^RY07] to prevent rogue-key attacks [^MOR01] where
+//! malicious signers adversarially pick their public keys in order to forge a multisignature.
+//!
+//! We implement the `Minimal-pubkey-size` variant from the BLS IETF draft standard [^bls-ietf-draft],
+//! which puts the signatures in the group $\mathbb{G}_2$ and the public keys in $\mathbb{G}_1$. The
+//! reasoning behind this choice is to minimize public key size, since public keys get posted on the
+//! blockchain.
+//!
+//! # Overview of BLS multisignatures
+//!
+//! In a _normal signature scheme_, we have a single _signer_ who generates its own key-pair:
+//! a _private-key_ and a corresponding _public key_. The signer can produce a _signature_ on a
+//! _message_ `m` using its private-key. Any _verifier_ who has the public key can check that
+//! the signature on `m` was produced by the signer.
+//!
+//! In a _multisignature scheme_, we have `n` signers. Each signer `i` has their own key-pair `(sk_i, pk_i)`.
+//! Any subset of `k` signers can collaborate to produce a succinct signature on the same message `m`.
+//! This is referred to as a _multisignature_ on `m`.
+//!
+//! Typically, the `k` signers first agree on the message `m` via some protocol (e.g., `m` is the
+//! latest block header in a blockchain protocol). Then, each signer produces a _signature share_ `s_i`
+//! on `m` using their own private key `sk_i`. After this, each signer `i` sends their signature
+//! share `s_i` to an _aggregator_: a dedicated, untrusted party who is responsible for aggregating
+//! the signature shares into the final multisignature. (For example, one of the signers themselves
+//! could be the aggregator.)
```

Comment:
We don't really need parenthesis here. Rewriting the final sentence as "For example, ... the aggregator." is more direct and better, I think.